public/app/types/unified-alerting-dto.ts

```diff
@@ -197,6 +197,7 @@ export interface GrafanaRuleDefinition extends PostableGrafanaRuleDefinition {
   namespace_uid: string;
   namespace_id: number;
   provenance?: string;
+  intervalSeconds?: number;
```

Comment:
Just curious, how (where in the code) does this field get its value?

Reply:
I'm new to the codebase but I think the location is here:

Inside the function `validateRuleNode` there is a parse using the golang time package:
```
intervalSeconds := int64(interval.Seconds())
```

I think this parse comes from the front-end field 'Interval' which is sent when we save/update the alert rule
So in the database the duration is saved in seconds in the field IntervalSeconds and when we retrieve the alert rule its called function `toGettableRuleGroupConfig` that parses back the time in seconds to string